### ISSUE TITLE
Linter output no longer shows up under WARN or INFO log levels

### ISSUE THREAD

**Issue by rmatte** [super-linter/super-linter#5218]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Linter output is not displayed under log level WARN or INFO (it was displayed under WARN in previous versions). It is now only displayed under the DEBUG log level.

### Expected Behavior

In previous versions super-linter would display the output from the linters in WARN log level so that we could see what the linters are complaining about and fix it. Now this information is only displayed under DEBUG logging level which produces tens of thousands or even hundreds of thousands of lines. It's impractical to have to sift through all of that just to figure out what's wrong every time we want to lint something.

### Super-Linter version

```markdown
I can't figure out how to get the version from inside the container and it doesn't seem to output
it when running. It's either 6.0.0 or 5.7.2. I pulled it down yesterday (02/01/2024) as the latest
docker image, so whatever version that would be.
```

### Relevant log output

```shell
Note that linters fail but don't show the output from the linters indicating what they are complaining about.
We use this linter automatically in places but we also use it as a manual tool for reviewing one off scripts
before they are submitted, and with the linter output missing from WARN level logging this is a major
issue for us. Scraping through a mountain of DEBUG log output to find the linter output is a waste of time.

Also, all of this linter version info being output just below the ascii art never used to be there when using
LOG_LEVEL=WARN. That all seems like it should only be output when LOG_LEVEL=INFO is in use.
LOG_LEVEL=WARN should only show linting failures and output from the linters like it used to
(as well as any fatal error conditions).

Logging levels in general seem to have gotten pretty messed up with this last update.

--------------------------------------------------------------------------------

                              /@@#///////@@/(@//@%/(@.@(       @@
                          @@//////////////////////////////#*  @@@
                        @////@//(///////////@@@@@///@//@/@**//@@(
                      @///////@///////////////@@@@    (           @,
                     @/(&/@////////////////////                     @
                    @////////////////////////@@                      @
                  @%////////(//////////%/////&@            @@       *,@           ______________
             @@@@@/@/#/////(&//////////////////                       .@         /              \
        *@@@@@.    .%///(//@//////////////////&.   .@@,                 @%      / Don't mind me  \
      @@%           .&@&&/@.@//&/////(//////////    @@@@@@@@@         .. &@    / I'm just looking \
    @@%               @@@@@   @&/////////////////#   @/       V  @@/ ,@@@ @   <  for some trash... |
@@@%                   @@@@        .%@@@@//////#@ @   @@         @     .,.     \__________________/
                                          @@@/@(  (@@@@% @/\      %
                                           @@@@(    .     .@@/\   #
                                             @                  %@%

--------------------------------------------------------------------------------
chktex: ChkTeX v1.7.8 - Copyright 1995-96 Jens T. Berger Thielemann.
Compiled with POSIX extended regex support.
rubocop: 1.59.0
perl:
This is perl 5, version 38, subversion 2 (v5.38.2) built for x86_64-linux-thread-multi

Copyright 1987-2023, Larry Wall

Perl may be copied only under the terms of either the Artistic License or the
GNU General Public License, which may be found in the Perl 5 source kit.

Complete documentation for Perl, including FAQ lists, should be found on
this system using "man perl" or "perldoc perl".  If you have access to the
Internet, point your browser at https://www.perl.org/, the Perl Home Page.
xmllint: xmllint: using libxml version 21106
   compiled with: Threads Tree Output Push Reader Patterns Writer SAXv1 HTTP DTDValid HTML C14N Catalog XPath XPointer XInclude Iconv ISO8859X Unicode Regexps Automata Schemas Schematron Modules Debug Zlib Lzma
ansible-lint: ansible-lint 6.22.2 using ansible-core:2.16.3 ansible-compat:4.1.11 ruamel-yaml:0.18.5 ruamel-yaml-clib:0.2.8
eslint: v8.56.0
markdownlint: 0.39.0
snakefmt: snakefmt, version 0.9.0
cpplint: Cpplint fork (https://github.com/cpplint/cpplint)
cpplint 1.6.1
Python 3.12.1 (main, Jan 27 2024, 06:36:14) [GCC 13.2.1 20231014]
editorconfig-checker: 2.7.2
gitleaks: v8.18.1
phpstan: PHPStan - PHP Static Analysis Tool 1.10.48
dotenv-linter: dotenv-linter 3.3.0
asl-validator: 3.8.2
actionlint: 1.6.26
installed by building from source
built with go1.21.1 compiler for linux/amd64
tekton-lint: 1.0.0-beta.9
terrascan: version: v1.18.11
terraform: Terraform v1.7.1
on linux_amd64
2024/01/30 20:02:15 traces export: Post "http:///dev/otel-grpc.sock": http: no Host in request URL
2024/01/30 20:02:15 traces export: Post "http:///dev/otel-grpc.sock": http: no Host in request URL
clippy: clippy 0.1.72
R: R version 4.3.1 (2023-06-16) -- "Beagle Scouts"
Copyright (C) 2023 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-musl (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under the terms of the
GNU General Public License versions 2 or 3.
For more information about these matters see
https://www.gnu.org/licenses/.
rustfmt: rustfmt 1.6.0-stable ( )
shellcheck: ShellCheck - shell script analysis tool
version: 0.9.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
jscpd: 3.5.10
black: black, 24.1.1 (compiled: no)
Python (CPython) 3.12.1
php: PHP 8.2.15 (cli) (built: Jan 18 2024 16:40:05) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.15, Copyright (c) Zend Technologies
textlint: v13.4.1
npm-groovy-lint: GroovyLint: Successfully processed CodeNarc:
CodeNarc version 3.3.0
npm-groovy-lint version 14.2.0

Embeds:
CodeNarc version 3.3.0
- Groovy version 3.0.9 (superlite)
renovate-config-validator: (node:399) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
37.161.0
golangci-lint: golangci-lint has version v1.55.2 built with go1.21.3 from e3c2265f on 2023-11-03T13:27:54Z
kubeconform: v0.6.4
golangci-lint: golangci-lint has version v1.55.2 built with go1.21.3 from e3c2265f on 2023-11-03T13:27:54Z
gherkin-lint: Version command not supported
eslint: v8.56.0
scalafmt: scalafmt 3.7.17
stylelint: 15.11.0
sqlfluff: sqlfluff, version 2.3.5
bash-exec: Version command not supported
google-java-format: google-java-format: Version 1.19.2
phpcs: PHP_CodeSniffer version 3.8.0 (stable) by Squiz and PHPCSStandards
terragrunt: terragrunt version v0.54.22
psalm: Psalm 5.20.0@3f284e96c9d9be6fe6b15c79416e1d1903dcfef4
pylint: pylint 3.0.3
astroid 3.0.2
Python 3.12.1 (main, Jan 27 2024, 06:36:14) [GCC 13.2.1 20231014]
shfmt: v3.7.0
snakemake: 8.4.1
arm-ttk: ModuleVersion = 0.24
pwsh: PowerShell 7.4.1
mypy: mypy 1.8.0 (compiled: yes)
eslint: v8.56.0
coffeelint: 5.2.11
tflint: TFLint version 0.50.2
+ ruleset.terraform (0.5.0-bundled)
cfn-lint: cfn-lint 0.85.0
flake8: 7.0.0 (mccabe: 0.7.0, pycodestyle: 2.11.1, pyflakes: 3.2.0) CPython 3.12.1 on
Linux
spectral: 6.11.0
clj-kondo: clj-kondo v2023.12.15
eslint: v8.56.0
ts-standard: 12.0.2
eslint: v8.56.0
protolint: protolint version 0.47.5(80ba08f)
raku: Welcome to Rakudo™ v2023.11.
Implementing the Raku® Programming Language v6.d.
Built on MoarVM version 2023.11.
hadolint: Haskell Dockerfile Linter 2.12.0
checkstyle: Checkstyle version: 10.13.0
clang-format: clang-format version 17.0.5 (https://github.com/llvm/llvm-project.git 98bfdac5ce82d1679f8af9a57501471812ab68d7)
ktlint: 1.1.1
yamllint: yamllint 1.33.0
dart: Dart SDK version: 3.2.5 (stable) (Tue Jan 16 15:02:13 2024 +0000) on "linux_x64"
isort:
                 _                 _
                (_) ___  ___  _ __| |_
                | |/ _/ / _ \/ '__  _/
                | |\__ \/\_\/| |  | |_
                |_|\___/\___/\_/   \_/

      isort your imports, so you don't have to.

                    VERSION 5.13.2
dotnet: 6.0.418
eslint: v8.56.0
rustfmt: rustfmt 1.6.0-stable ( )
htmlhint: 1.1.4
rustfmt: rustfmt 1.6.0-stable ( )
sql-lint: 1.0.0
checkov: 3.2.1
lua: Lua 5.3.5  Copyright (C) 1994-2018 Lua.org, PUC-Rio
standard: 17.1.0
2024-02-02 20:33:08 [ERROR]   Errors found in PYTHON_BLACK
2024-02-02 20:33:08 [ERROR]   Errors found in PYTHON_FLAKE8
2024-02-02 20:33:08 [ERROR]   Super-linter detected linting errors
```

### Steps To Reproduce

Run an older version of super-linter and lint some code which fails linting with LOG_LEVEL=WARN enabled, notice that it shows you the output from the linter indicating what it's not liking. Then run the latest version of super-linter in the same way and notice that it'll tell you that linters failed but it won't show you what they are complaining about like it did in previous versions. 

### Anything else?

With LOG_LEVEL=WARN it used to only output the ascii art at the top and any linting specific errors. Now it floods the screen with a bunch of linter version info which seems informational and should probably only be output when LOG_LEVEL=INFO is in use. It seems like log levels got messed up in general in this latest update.

**Comment by ferrarimarco** [comment]
Hi @rmatte !

Linter output is there at the INFO level. If you set this to anything higher, it doesn't show up. What happens if you don't set the log level?

**Comment by rmatte** [comment]
> Hi @rmatte !
> 
> Linter output is there at the INFO level. If you set this to anything higher, it doesn't show up. What happens if you don't set the log level?

Here is a run with LOG_LEVEL=INFO (note that the WARN lines for failed linting that I get when I run with LOG_LEVEL=WARN are not missing under LOG_LEVEL=INFO for whatever reason, which seems very odd, it's like it's working backwards from the way it should be.

If I run it with no LOG_LEVEL defined at all then I get the general output that I'd expect to see. So I guess for now we'll just need to not explicitly set the LOG_LEVEL, but this still seems like the LOG_LEVEL directive itself is behaving strangely.

Thanks for the workaround.

```
[rmatte@rmatte lintme]$ docker run -e USE_FIND_ALGORITHM=true -e FILTER_REGEX_EXCLUDE=.mypy_cache/.* -e RUN_LOCAL=true -e LOG_LEVEL=INFO -e PERL_PERLCRITIC_OPTIONS='--theme community --verbose 5' -v ~/lintme:/tmp/lint ghcr.io/super-linter/super-linter:latest
--------------------------------------------------------------------------------

                              /@@#///////@@/(@//@%/(@.@(       @@
                          @@//////////////////////////////#*  @@@
                        @////@//(///////////@@@@@///@//@/@**//@@(
                      @///////@///////////////@@@@    (           @,
                     @/(&/@////////////////////                     @
                    @////////////////////////@@                      @
                  @%////////(//////////%/////&@            @@       *,@           ______________
             @@@@@/@/#/////(&//////////////////                       .@         /              \
        *@@@@@.    .%///(//@//////////////////&.   .@@,                 @%      / Don't mind me  \
      @@%           .&@&&/@.@//&/////(//////////    @@@@@@@@@         .. &@    / I'm just looking \
    @@%               @@@@@   @&/////////////////#   @/       V  @@/ ,@@@ @   <  for some trash... |
@@@%                   @@@@        .%@@@@//////#@ @   @@         @     .,.     \__________________/
                                          @@@/@(  (@@@@% @/\      %
                                           @@@@(    .     .@@/\   #
                                             @                  %@%

--------------------------------------------------------------------------------
chktex: ChkTeX v1.7.8 - Copyright 1995-96 Jens T. Berger Thielemann.
Compiled with POSIX extended regex support.
rubocop: 1.59.0
perl:
This is perl 5, version 38, subversion 2 (v5.38.2) built for x86_64-linux-thread-multi

Copyright 1987-2023, Larry Wall

Perl may be copied only under the terms of either the Artistic License or the
GNU General Public License, which may be found in the Perl 5 source kit.

Complete documentation for Perl, including FAQ lists, should be found on
this system using "man perl" or "perldoc perl".  If you have access to the
Internet, point your browser at https://www.perl.org/, the Perl Home Page.
xmllint: xmllint: using libxml version 21106
   compiled with: Threads Tree Output Push Reader Patterns Writer SAXv1 HTTP DTDValid HTML C14N Catalog XPath XPointer XInclude Iconv ISO8859X Unicode Regexps Automata Schemas Schematron Modules Debug Zlib Lzma
ansible-lint: ansible-lint 6.22.2 using ansible-core:2.16.3 ansible-compat:4.1.11 ruamel-yaml:0.18.5 ruamel-yaml-clib:0.2.8
eslint: v8.56.0
markdownlint: 0.39.0
snakefmt: snakefmt, version 0.9.0
cpplint: Cpplint fork (https://github.com/cpplint/cpplint)
cpplint 1.6.1
Python 3.12.1 (main, Jan 27 2024, 06:36:14) [GCC 13.2.1 20231014]
editorconfig-checker: 2.7.2
gitleaks: v8.18.1
phpstan: PHPStan - PHP Static Analysis Tool 1.10.48
dotenv-linter: dotenv-linter 3.3.0
asl-validator: 3.8.2
actionlint: 1.6.26
installed by building from source
built with go1.21.1 compiler for linux/amd64
tekton-lint: 1.0.0-beta.9
terrascan: version: v1.18.11
terraform: Terraform v1.7.1
on linux_amd64
2024/01/30 20:02:15 traces export: Post "http:///dev/otel-grpc.sock": http: no Host in request URL
2024/01/30 20:02:15 traces export: Post "http:///dev/otel-grpc.sock": http: no Host in request URL
clippy: clippy 0.1.72
R: R version 4.3.1 (2023-06-16) -- "Beagle Scouts"
Copyright (C) 2023 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-musl (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under the terms of the
GNU General Public License versions 2 or 3.
For more information about these matters see
https://www.gnu.org/licenses/.
rustfmt: rustfmt 1.6.0-stable ( )
shellcheck: ShellCheck - shell script analysis tool
version: 0.9.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
jscpd: 3.5.10
black: black, 24.1.1 (compiled: no)
Python (CPython) 3.12.1
php: PHP 8.2.15 (cli) (built: Jan 18 2024 16:40:05) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.15, Copyright (c) Zend Technologies
textlint: v13.4.1
npm-groovy-lint: GroovyLint: Successfully processed CodeNarc:
CodeNarc version 3.3.0
npm-groovy-lint version 14.2.0

Embeds:
CodeNarc version 3.3.0
- Groovy version 3.0.9 (superlite)
renovate-config-validator: (node:399) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
37.161.0
golangci-lint: golangci-lint has version v1.55.2 built with go1.21.3 from e3c2265f on 2023-11-03T13:27:54Z
kubeconform: v0.6.4
golangci-lint: golangci-lint has version v1.55.2 built with go1.21.3 from e3c2265f on 2023-11-03T13:27:54Z
gherkin-lint: Version command not supported
eslint: v8.56.0
scalafmt: scalafmt 3.7.17
stylelint: 15.11.0
sqlfluff: sqlfluff, version 2.3.5
bash-exec: Version command not supported
google-java-format: google-java-format: Version 1.19.2
phpcs: PHP_CodeSniffer version 3.8.0 (stable) by Squiz and PHPCSStandards
terragrunt: terragrunt version v0.54.22
psalm: Psalm 5.20.0@3f284e96c9d9be6fe6b15c79416e1d1903dcfef4
pylint: pylint 3.0.3
astroid 3.0.2
Python 3.12.1 (main, Jan 27 2024, 06:36:14) [GCC 13.2.1 20231014]
shfmt: v3.7.0
snakemake: 8.4.1
arm-ttk: ModuleVersion = 0.24
pwsh: PowerShell 7.4.1
mypy: mypy 1.8.0 (compiled: yes)
eslint: v8.56.0
coffeelint: 5.2.11
tflint: TFLint version 0.50.2
+ ruleset.terraform (0.5.0-bundled)
cfn-lint: cfn-lint 0.85.0
flake8: 7.0.0 (mccabe: 0.7.0, pycodestyle: 2.11.1, pyflakes: 3.2.0) CPython 3.12.1 on
Linux
spectral: 6.11.0
clj-kondo: clj-kondo v2023.12.15
eslint: v8.56.0
ts-standard: 12.0.2
eslint: v8.56.0
protolint: protolint version 0.47.5(80ba08f)
raku: Welcome to Rakudo™ v2023.11.
Implementing the Raku® Programming Language v6.d.
Built on MoarVM version 2023.11.
hadolint: Haskell Dockerfile Linter 2.12.0
checkstyle: Checkstyle version: 10.13.0
clang-format: clang-format version 17.0.5 (https://github.com/llvm/llvm-project.git 98bfdac5ce82d1679f8af9a57501471812ab68d7)
ktlint: 1.1.1
yamllint: yamllint 1.33.0
dart: Dart SDK version: 3.2.5 (stable) (Tue Jan 16 15:02:13 2024 +0000) on "linux_x64"
isort:
                 _                 _
                (_) ___  ___  _ __| |_
                | |/ _/ / _ \/ '__  _/
                | |\__ \/\_\/| |  | |_
                |_|\___/\___/\_/   \_/

      isort your imports, so you don't have to.

                    VERSION 5.13.2
dotnet: 6.0.418
eslint: v8.56.0
rustfmt: rustfmt 1.6.0-stable ( )
htmlhint: 1.1.4
rustfmt: rustfmt 1.6.0-stable ( )
sql-lint: 1.0.0
checkov: 3.2.1
lua: Lua 5.3.5  Copyright (C) 1994-2018 Lua.org, PUC-Rio
standard: 17.1.0
```

**Comment by ferrarimarco** [comment]
Can you try by actually removing the LOG_LEVEL variable entirely? The result you're getting is quite surprising

**Comment by rmatte** [comment]
Just to summarize this...

LOG_LEVEL=WARN displays errors like this but no linter output (we used to see linter output when running with WARN but we don't anymore).

2024-02-02 21:36:17 [ERROR]   Errors found in PYTHON_BLACK
2024-02-02 21:36:17 [ERROR]   Errors found in PYTHON_FLAKE8
2024-02-02 21:36:17 [ERROR]   Super-linter detected linting errors

LOG_LEVEL=INFO doesn't display the above ERROR lines at all, even though I'm linting the same file with it and it should be displaying those.

Not setting LOG_LEVEL at all shows me all the stuff I want to see so we'll just use that as a workaround for now on our end.

I would expected a LOG_LEVEL=INFO to display anything INFO or above but those ERROR lines above get hidden when I run it that way. There seems to be some sort of weird bug here that's probably worth investigating.

**Comment by ferrarimarco** [comment]
@grahame-student reported a similar request in #5215, although it ended up being a different issue in the end. This is probably a fair request. I'll happily review a PR for this.

**Comment by ferrarimarco** [comment]
By the way (somewhat surprisingly), `INFO` is not currently a valid log level. The setting you're looking for is `LOG_LEVEL=VERBOSE`. We should probably update this, but that's a separate issue.

**Comment by grahame-student** [comment]
> @grahame-student reported a similar request in #5215, although it ended up being a different issue in the end. This is probably a fair request. I'll happily review a PR for this.

The observation I made after #5215 was closed is still relevant to this issue I think.
Basically I noted that with the log level set to NOTICE I wasn't seeing checkov's output, just the summary that it had spotted an issue.

**Comment by ferrarimarco** [comment]
@grahame-student When we get to work on this (PRs are welcome :) we'll solve it for `NOTICE` as well.

Or was your observation about something else?

**Comment by grahame-student** [comment]
@ferrarimarco I think our understanding on this is now aligned, thanks.

**Comment by ferrarimarco** [comment]
#5251 should take care of this.

**Comment by rmatte** [comment]
@ferrarimarco Now with the latest version no matter whether or not I run it with the LOG_LEVEL option it's always showing me DEBUG output, for instance all of these commands show DEBUG log output...

docker run -e USE_FIND_ALGORITHM=true -e FILTER_REGEX_EXCLUDE=.mypy_cache/.* -e RUN_LOCAL=true -e PERL_PERLCRITIC_OPTIONS='--theme community --verbose 5' -v ~/lintme:/tmp/lint ghcr.io/super-linter/super-linter:latest

docker run -e LOG_LEVEL=INFO -e USE_FIND_ALGORITHM=true -e FILTER_REGEX_EXCLUDE=.mypy_cache/.* -e RUN_LOCAL=true -e PERL_PERLCRITIC_OPTIONS='--theme community --verbose 5' -v ~/lintme:/tmp/lint ghcr.io/super-linter/super-linter:latest

docker run -e LOG_LEVEL=WARN -e USE_FIND_ALGORITHM=true -e FILTER_REGEX_EXCLUDE=.mypy_cache/.* -e RUN_LOCAL=true -e PERL_PERLCRITIC_OPTIONS='--theme community --verbose 5' -v ~/lintme:/tmp/lint ghcr.io/super-linter/super-linter:latest

docker run -e LOG_LEVEL=ERROR -e USE_FIND_ALGORITHM=true -e FILTER_REGEX_EXCLUDE=.mypy_cache/.* -e RUN_LOCAL=true -e PERL_PERLCRITIC_OPTIONS='--theme community --verbose 5' -v ~/lintme:/tmp/lint ghcr.io/super-linter/super-linter:latest

The --verbose 5 under the perlcritic options has nothing to do with the issue, I tried removing that and I get the same behavior. This is making it very difficult for us to visually spot issues while running superlinter because we need to wade through all of this debug log output.

**Comment by ferrarimarco** [comment]
@rmatte I think you're right, there's something off. I'll have a look.

**Comment by rmatte** [comment]
> @rmatte I think you're right, there's something off. I'll have a look.

Seems related to this: https://github.com/super-linter/super-linter/commit/0f91a56f2180b245f2876612816410a5dc73bdf3

**Comment by ferrarimarco** [comment]
@rmatte #5319 should take care of this.
